plugins/available/alias-completion.plugin.bash

```diff
@@ -17,54 +18,58 @@ about-plugin 'Automatic completion of aliases'
+	# read in "<alias> '<aliased command>' '<command args>'" lines from defined aliases
+	local line
+	while read -r line; do
+		eval "local alias_tokens; alias_tokens=($line)" 2> /dev/null || continue # some alias arg patterns cause an eval parse error
+		local alias_name="${alias_tokens[0]}" alias_cmd="${alias_tokens[1]}" alias_args="${alias_tokens[2]# }"
 
-        # skip aliases to pipes, boolean control structures and other command lists
-        # (leveraging that eval errs out if $alias_args contains unquoted shell metacharacters)
-        eval "local alias_arg_words; alias_arg_words=($alias_args)" 2>/dev/null || continue
-        # avoid expanding wildcards
-        read -a alias_arg_words <<< "$alias_args"
+		# skip aliases to pipes, boolean control structures and other command lists
+		# (leveraging that eval errs out if $alias_args contains unquoted shell metacharacters)
+		eval "local alias_arg_words; alias_arg_words=($alias_args)" 2> /dev/null || continue
+		# avoid expanding wildcards
+		read -ra alias_arg_words <<< "$alias_args"
 
-        # skip alias if there is no completion function triggered by the aliased command
-        if [[ ! " ${completions[*]} " =~ " $alias_cmd " ]]; then
-            if [[ -n "$completion_loader" ]]; then
-                # force loading of completions for the aliased command
-                eval "$completion_loader $alias_cmd"
-                # 124 means completion loader was successful
-                [[ $? -eq 124 ]] || continue
-                completions+=($alias_cmd)
-            else
-                continue
-            fi
-        fi
-        local new_completion="$(complete -p "$alias_cmd" 2>/dev/null)"
+		# skip alias if there is no completion function triggered by the aliased command
+		if [[ ! " ${completions[*]} " =~ $alias_cmd ]]; then
+			if [[ -n "$completion_loader" ]]; then
+				# force loading of completions for the aliased command
+				eval "$completion_loader $alias_cmd"
+				# 124 means completion loader was successful
+				[[ $? -eq 124 ]] || continue
+				completions+=("$alias_cmd")
+			else
+				continue
+			fi
+		fi
+		local new_completion="$(complete -p "$alias_cmd" 2> /dev/null)"
 
-        # create a wrapper inserting the alias arguments if any
-        if [[ -n $alias_args ]]; then
-            local compl_func="${new_completion/#* -F /}"; compl_func="${compl_func%% *}"
-            # avoid recursive call loops by ignoring our own functions
-            if [[ "${compl_func#_$namespace::}" == $compl_func ]]; then
-                local compl_wrapper="_${namespace}::${alias_name}"
-                    echo "function $compl_wrapper {
+		# create a wrapper inserting the alias arguments if any
+		if [[ -n $alias_args ]]; then
+			local compl_func="${new_completion/#* -F /}"
+			compl_func="${compl_func%% *}"
+			# avoid recursive call loops by ignoring our own functions
+			if [[ "${compl_func#_$namespace::}" == "$compl_func" ]]; then
```

Comment:
So shellcheck recommends **no** quotes for `=~` ( ie `=~ $alias_cmd` above) but _does_ recommend quotes for `==` ?

Reply:
Shellcheck do recommends it as the `=~` uses a regexp and the latter uses a literal comparison. I'll check if there's a use case when this could be ignored as it is or if we should follow shellcheck